processing/src/main/java/io/druid/query/groupby/GroupByQuery.java

```diff
@@ -819,17 +678,21 @@ public Builder setContext(Map<String, Object> context)
       return this;
     }
 
+    public Builder overrideContext(Map<String, Object> contextOverride)
+    {
+      this.context = computeOverriddenContext(context, contextOverride);
+      return this;
+    }
+
     public Builder setHavingSpec(HavingSpec havingSpec)
     {
       this.havingSpec = havingSpec;
-
       return this;
     }
 
     public Builder setLimit(Integer limit)
     {
       this.limit = limit;
```

Comment:
i think its would be less error prone to just remove `limitSpec == null` check and recreate it every time.